hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/AccountAliasesValidator.java

```diff
@@ -0,0 +1,114 @@
+
+public class AccountAliasesValidator implements RecordStreamValidator {
+    private final Map<ByteString, Long> aliasesFromRecords = new HashMap<>();
+    private final Map<ByteString, Long> aliasesFromState = new HashMap<>();
+
+    @Override
+    @SuppressWarnings("java:S4507")
+    public void validateRecordsAndSidecars(@NonNull final List<RecordWithSidecars> recordsWithSidecars) {
+        requireNonNull(recordsWithSidecars);
+        getExpectedAliasesFrom(recordsWithSidecars);
+
+        new SimpleFreezeOnly().runSuiteSync(); // freeze the network before reading the signed state file
+        try (Stream<Path> paths =
+                Files.walk(Paths.get("build/network/itest/saved/node_0/com.hedera.node.app.ServicesMain/0/hedera"))) {
+            var signedStateFile = Streams.findLast(
+                            paths.filter(p -> p.endsWith("SignedState.swh")).sorted())
+                    .orElseThrow(); // get the latest signed state file
+            getActualAliasesFrom(signedStateFile);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        if (!aliasesFromRecords.equals(aliasesFromState)) {
+            throw new IllegalStateException("Expected " + aliasesFromRecords.size() + " aliases from records, but got "
+                    + aliasesFromState.size() + "from state");
+        }
+    }
+
+    private void getExpectedAliasesFrom(final List<RecordWithSidecars> recordsWithSidecars) {
+        for (final var recordWithSidecars : recordsWithSidecars) {
+            final var items = recordWithSidecars.recordFile().getRecordStreamItemsList();
+            for (final var item : items) {
+                if (item.getRecord().getReceipt().getStatus() != ResponseCodeEnum.SUCCESS) {
+                    continue;
+                }
+
+                try {
+                    final var txn = CommonUtils.extractTransactionBody(item.getTransaction());
+                    var alias = ByteString.EMPTY;
+                    if (txn.hasCryptoCreateAccount()) {
+                        alias = txn.getCryptoCreateAccount().getAlias();
+                    }
+
+                    var accountNum =
+                            item.getRecord().getReceipt().getAccountID().getAccountNum();
+                    if (!alias.isEmpty()) {
+                        aliasesFromRecords.put(alias, accountNum);
+                    }
+
+                    var evmAddressAlias = item.getRecord().getEvmAddress();
+                    if (!evmAddressAlias.isEmpty()) {
+                        aliasesFromRecords.put(evmAddressAlias, accountNum);
+                    }
+                } catch (final InvalidProtocolBufferException e) {
+                    throw new IllegalStateException(e);
+                }
+            }
+        }
+    }
+
+    private void getActualAliasesFrom(final Path signedStateFile) throws Exception {
+        var state = new SignedStateHolder(signedStateFile).getPlatformState();
```

Comment:
This won't work in general: The platform state is only valid while the `SignedStateHolder` exists.  A `SignedStateHolder` is an `AutoCloseable` so hold it in a try-with-resources.